packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -117,4 +119,73 @@ export class HttpStage extends Resource implements IStage {
     const urlPath = this.stageName === DEFAULT_STAGE_NAME ? '' : this.stageName;
     return `https://${this.httpApi.httpApiId}.execute-api.${s.region}.${s.urlSuffix}/${urlPath}`;
   }
+
+  /**
+   * Return the given named metric for this HTTP Api Gateway Stage
+   *
+   * @default Average over 5 minutes
+   */
+  public metric(metricName: string, props?: MetricOptions): Metric {
+    var api = this.httpApi as HttpApi;
```

Comment:
This is not guaranteed to be true. It can also be an imported HttpApi, in which case, this will fail with an error.

It's safe to elevate the `metric*()` APIs to the `IHttpApi` interface and introducing an `HttpApiBase` class. This way the logic will work for both imported and CDK implemented `IHttpApi`.

Change the code like so -

```ts
export interface IHttpApi extends IResource {
  // ... what's already there

  metric(metricName: string, props?: MetricOptions): Metric;

  metricClientError(props?: MetricOptions): Metric;

  // ...
}

class HttpApiBase extends Resource implements IHttpApi { // note that this is not exported
  public metric(metricName: string, props?: MetricOptions): Metric {
    // ...
  }

  public metricClientError(props?: MetricOptions): Metric {
    // ...
  }
}

export class HttpApi extends HttpApiBase {
  public static fromApiId(scope: Construct, id: string, httpApiId: string): IHttpApi {
    class Import extends HttpApiBase {
      public readonly httpApiId = httpApiId;
    }
    return new Import(scope, id);
  }

  // ...
}
```

Reply:
HttpApiBase created and metrics moved to interface.